ui/server/src/main/scala/pl/touk/nussknacker/ui/api/ManagementResources.scala

```diff
@@ -122,11 +128,16 @@ class ManagementResources(processCounter: ProcessCounter,
   private implicit final val plainBytes: FromEntityUnmarshaller[Array[Byte]] = Unmarshaller.byteArrayUnmarshaller
   private implicit final val plainString: FromEntityUnmarshaller[String] = Unmarshaller.stringUnmarshaller
 
-  private def withComment: Directive1[Option[String]] =
-    entity(as[Option[String]]).map(_.filterNot(_.isEmpty)).flatMap {
-      case None if deploySettings.exists(_.requireComment) => reject(ValidationRejection("Comment is required", None))
-      case comment => provide(comment)
+  case class ValidationError(message: String) extends Exception(message) with BadRequestError
+
+  private def withDeploymentComment: Directive1[Option[DeploymentComment]] = {
+    entity(as[Option[String]]).flatMap{ comment =>
+      DeploymentComment.maybeDeploymentComment(comment, deploySettings) match {
```

Comment:
You haven't pushed this method...

Reply:
as above, renamed definition after review, didn't rename implementation